### ISSUE TITLE
Can't get Maigret to work on Tails over Tor?

### ISSUE THREAD

**Issue by S0yf00d-exe** [soxoj/maigret#544]
## Checklist

- ~~[ ]~~ I'm reporting a bug in Maigret functionality
- ~~[ ]~~ I've checked for similar bug reports including closed ones
- [ ] I've checked for pull requests that attempt to fix this bug

## Description

<**`Version --v0.4.3`,** **`TailsOS`** >

<**Can't Get Tor to work with Maigret on Tails**>

<PASTE SCREENSHOT>

<ATTACH LOG FILE>
So I have Tails setup with persistence and installed all required depencies like *Python 3.9* and *Pipe3*. Then tried installing it did seem to work but then running `torsocks maigret specified username` it came back with command not found.

Therefore I went the clone route: 

`git clone https://github.com/soxoj/maigret && cd maigret`

`pip3 install -r requirements.tx`

Which then worked when doing `torsocks` `./maigret.py specified username`. 

However it would try loading up *torsocks* connection would drop and keep trying and I checked my *Onion circuits* and indeed it was pinging the sites in list.

At the end it would say nothing found even when adding `--html` `--pdf` flags.

**What am I doing wrong as I heard this works fine with Tor and has Tor support?**

Tails used to use a Transparent Proxy for Tor connections, but not anymore as far as I know...

>https://tails.boum.org/contribute/design/Tor_enforcement/

I have tried `torsocks` and `torify` with no luck, without those it would say unable to connect as it needs to be told to go over Tor.

Is there something I did wrong when installing or a *Maigret* config change I need to do?

**Comment by fen0s** [comment]
maigret doesn't really support tor connections yet, and i'm not sure if Soxoj is up to implementing it
it might change in the future though

**Comment by S0yf00d-exe** [comment]
@fen0s

Well I followed what was mentioned in https://github.com/soxoj/maigret/issues/514

`./maigret.py` username `socks5://localhost:9050`

`/maigret.py` username `--html --pdf socks5://localhost:9050 --print-errors`

`/maigret.py` username `-a --html --pdf socks5://localhost:9050 --print-errors`

all these worked after adding

**`socks5://localhost:9050`** should be documented more:

>https://maigret.readthedocs.io/en/latest/command-line-options.html

However I had to reinstall the requirements again `torsocks pip3 install -r requirements.txt` before trying out getting Maigret to work again (before doing commands above).

**Comment by S0yf00d-exe** [comment]
@fen0s

**How can I fix this** **so I don't have to install the** `requirements.txt` **each time?**

![warning](https://user-images.githubusercontent.com/104031749/179152777-3e778476-d184-4f25-807d-0fbea3247c92.jpg)

**Comment by S0yf00d-exe** [comment]
@fen0s @soxoj

**Also regarding Tor what would be a proper** `--timeout` **command to not get as many timeout error requests?**

![timeout](https://user-images.githubusercontent.com/104031749/179153025-786aaa1e-4513-472c-8748-0e4fea09b31b.jpg)

**Comment by fen0s** [comment]
> @fen0s @soxoj
> 
> **Also regarding Tor what would be a proper** `--timeout` **command to not get as many timeout error requests?**
> 
> ![timeout](https://user-images.githubusercontent.com/104031749/179153025-786aaa1e-4513-472c-8748-0e4fea09b31b.jpg)

no one can't know for sure, but considering tor could be very slow, i'd say about 30

**Comment by fen0s** [comment]
> @fen0s
> 
> **How can I fix this** **so I don't have to install the** `requirements.txt` **each time?**
this one is probably connected to tails, maybe package installation scripts are going wrong somewhere? is there a more detailed log you can provide?
> ![warning](https://user-images.githubusercontent.com/104031749/179152777-3e778476-d184-4f25-807d-0fbea3247c92.jpg)

**Comment by S0yf00d-exe** [comment]
@fen0s
*So I tried installing it again with pip3.* 

I get this **[Warning/Error]** at the end when installing with pip3 method.
![error](https://user-images.githubusercontent.com/104031749/179429208-2a1fe921-f55a-492e-ba75-3f5d2bb48c22.jpg)

Then typing `maigret` says this command doesn't exist... but does work with 

`python3 -m maigret username --html --pdf --proxy socks5://localhost:9050`

*not sure were I should move to fix path*

What I have found is maigret is located in `/home/amnesia/.local/bin/` or `/.local/bin/` 

and the requirements are in `/home/amnesia/.local/lib/python3.9/site-packages/` or`/.local/lib/python3.9/site-packages/`

Both these folders `/home/amnesia/.local/bin/` & `/home/amnesia/.local/lib/` disapear in tails with persistence enabled.

>this one is probably connected to tails, maybe package installation scripts are going wrong somewhere?

### So I need to figure out how to make a dotfile to persist this but not sure what method route to go?

**Comment by fen0s** [comment]
sorry, i don't have experience using amnesiac OS, so i'm not sure how to solve this
you could just put those folders somewhere where persistence works and use them in your path variable?

**Comment by S0yf00d-exe** [comment]
@fen0s

Yeah I tried persisting `~/.local/bin/` & `/.local/lib/` as a dotfile but broke my tails however I was able to unbrick it by using another USB and deleteing that dotfile in the persistent volume.

Yeah idk if there is a sure way with tails maybe autorun script that installs it each time idk.

You can close this issue if you like but I will say definitly add documentation about: 

**Using Tor Proxy** 
`--proxy socks5://localhost:9050` and other port `--proxy socks5://localhost:9150`

**Comment by soxoj** [comment]
Documentation was updated in #2663.
